### PR TITLE
Automatically generate culprits when missing

### PR DESCRIPTION
**Pull request by dcramer**
Attempts to find the best frame by looking for the most recent frame marked as in_app, or the most recent frame if not present.

Refs GH-1989

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -109,6 +110,30 @@ def should_sample(current_datetime, last_seen, times_seen):
...
 
 
+def generate_culprit(data):
+    from sentry.interfaces.stacktrace import Stacktrace
+
+    try:
+        stacktraces = [
+            e['stacktrace']
+            for e in data['sentry.interfaces.Exception']['values']
+            if e.get('stacktrace')
```
Comment (mattrobenolt) [inline src/sentry/event_manager.py]: `if 'stacktrace' in e`

**Comment by mattrobenolt** [comment]
:shipit:

File: src/sentry/event_manager.py (hunk 1 above)
Comment (dcramer) [inline src/sentry/event_manager.py]: stacktrace can be empty

Comment (mattrobenolt) [inline src/sentry/event_manager.py]: Ah, I thought you were checking if the key existed. My b.

**Pull request merged**